app/models/invite_request.rb

```diff
@@ -14,6 +16,20 @@ def self.reset_order
     end
   end
 
+  # Borrow the blacklist cleaner but just strip out all the periods for all domains
+  def set_simplified_email
+    return if email.blank?
+    simplified = AdminBlacklistedEmail.canonical_email(email).split('@')
+    self.simplified_email = simplified.first.delete(".").gsub(/\+.+$/, "") + "@#{simplified.last}"
+  end
+
+  # Doing this with a method so the error message makes more sense
+  def simplified_email_uniqueness
+    if InviteRequest.where(simplified_email: simplified_email).exists?
```

Comment:
Having the check be implemented like this means that all InviteRequests will become invalid immediately after being saved to database, which I'm pretty sure will break InviteRequest.reset_order. I'd suggest either adding `on: :create` to the validate statement above, or adding `.where.not(id: self.id)` here.

Reply:
Thanks! Switched from a regular uniqueness validation and didn't notice that.